frontend/src/Pages/MyProfilePage.js

```diff
@@ -74,6 +84,20 @@ export default function MyProfilePage() {
     }
   };
 
+  const handleEmailSave = async () => {
+    try {
+		await axios.patch(`${API_BASE_URL}/api/profile/change-email`, formData, {
+			withCredentials: true,
+		});
+
+
+      setProfile((prevProfile) => ({ ...prevProfile, ...formData }));
+      handleClose();
+    } catch (error) {
+      console.error("Error saving profile data", error);
```

Comment:
It is good practice to show error message to user to let him know that something went wrong. try to use https://mui.com/material-ui/react-alert/?srsltid=AfmBOoqSmIKu_9VX_e7nR9peH2LmeaQEeqzpsKUDifcq1DtgqH53dY3Y for displayin errors